common/models/transducer/transducer_training_pipeline/pipeline.py

```diff
@@ -0,0 +1,167 @@
+    self.stage = stage_list[0]  # saves the stage we are on
+    self.index = 0  # index of current stage
+    self.start_epoch = 1  # holds the epoch, the current stage started.
+    self.align_dir = os.path.dirname(get_global_config().value("model", "net-model/network"))
+    self.stage_list = stage_list
+    self._proof_check_stages()
+
+  def _proof_check_stages(self):
+    for (i, st) in enumerate(self.stage_list):
+      if st.fixed_path:
+        assert st.stage_num_align >= 0, f"The stage to take the alginments from is not set in stage {i}."
+
+  def _stage_epoch(self, epoch) -> int:
+    """Returns the epoch number relative to the start of current stage"""
+    return epoch - self.start_epoch
+
+  def _update(self, epoch: int):
+    """Update model for the next stage if necessary"""
+    if len(self.stage_list) > self.index and self.stage.num_epochs < self._stage_epoch(epoch):
+      self.index += 1
+      self.stage = self.stage_list[self.index]
+
+      self.start_epoch = epoch
+
+  def _get_net(self, epoch: int) -> Dict[str, Any]:
+    return self.stage.make_net(epoch)
+
+  def _get_net_with_align_dumping(self, epoch: int, ctx: Context) -> Dict[str, Any]:
+    net = self._get_net(epoch)
+    net = update_net_for_alignment_dumping(net=net, extend=False, ctx=ctx,
+                                           stage_num=self.index, align_dir=self.align_dir,
+                                           alignment_topology=self.stage.alignment_topology)
+    return net
+
+  def _get_net_with_fixed_path_training(self, epoch: int, ctx: Context) -> Dict[str, Any]:
+    net = self._get_net(epoch)
+    net = update_net_for_fixed_path_training(net=net, ctx=ctx, align_dir=self.align_dir,
+                                             stage_num_align=self.stage.stage_num_align)
+
+    # Global changes
+    # Reset
+    if self.stage.reset:
+      net["#copy_param_mode"] = "reset"
+
+    # Chunking
+    if self.stage.chunking:
+      _time_red = 6
+      _chunk_size = 60
+      net["#config"].update({
+        # TODO: more? e.g. maximize GPU mem util
+        "chunking":  # can use chunking with frame-wise training
+        (
+          {"data": _chunk_size * _time_red, "alignment": _chunk_size},
+          {"data": _chunk_size * _time_red // 2, "alignment": _chunk_size // 2}
+        )
+      })
+
+    return net
+
+  def get_network(self, epoch: int) -> Dict[str, Any]:
+    """Gets the network from the pretrainer
+    Builds and updates the network according to the epoch we are in now.
+    It adds alignment if required.
+    """
+
+
+    task = get_global_config().value("task", "train")
+    target = TargetConfig.global_from_config()
+    ctx = Context(task=task, target=target, beam_size=12)
+
+    self._update(epoch)
+    if self.stage.num_epochs == self._stage_epoch(epoch):  # create alignments
+      self.type = "Align"
+      net = self._get_net_with_align_dumping(epoch, ctx)
```

Comment:
Just noticed. this is wrong for the moment it only does align for 1 sub_epoch. Normally, it should do that for EpochSplit sub_epochs.

Reply:
And also, starting at some epoch with `epoch0 % EpochSplit == 0`, such that you have covered the full dataset.

Reply:
Yes, corrected it.